x-pack/test/apm_api_integration/tests/status_check/agent_status_check.spec.ts

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from '@kbn/expect';
+import { ApmSynthtraceEsClient } from '@kbn/apm-synthtrace';
+import { apm, timerange } from '@kbn/apm-synthtrace-client';
+import { FtrProviderContext } from '../../common/ftr_provider_context';
+
+async function generateData({
+  synthtraceEsClient,
+  start,
+  end,
+  type,
+}: {
+  synthtraceEsClient: ApmSynthtraceEsClient;
+  start: number;
+  end: number;
+  type: 'transaction' | 'error' | 'metric';
+}) {
+  const serviceName = 'synth-go';
+  const serviceGoProdInstance = apm
+    .service({ name: serviceName, environment: 'production', agentName: 'go' })
+    .instance('instance-a');
+
+  const metricOnlyInstance = apm
+    .service({ name: serviceName, environment: 'production', agentName: 'java' })
+    .instance('metric-only-production');
+
+  const transactionNameProductList = 'GET /api/product/list';
+
+  switch (type) {
+    case 'transaction':
+      await synthtraceEsClient.index(
+        timerange(start, end)
+          .interval('1m')
+          .rate(1)
+          .generator((timestamp) =>
+            serviceGoProdInstance
+              .transaction({ transactionName: transactionNameProductList })
+              .timestamp(timestamp)
+              .duration(1000)
+              .success()
+          )
+      );
+      break;
+    case 'error':
+      await synthtraceEsClient.index(
+        timerange(start, end)
+          .interval('1m')
+          .rate(1)
+          .generator((timestamp) =>
+            serviceGoProdInstance
+              .transaction({ transactionName: transactionNameProductList })
+              .timestamp(timestamp)
+              .duration(1000)
+              .failure()
+          )
+      );
+      break;
+    case 'metric':
+      await synthtraceEsClient.index(
+        timerange(start, end)
+          .interval('1m')
+          .rate(1)
+          .generator((timestamp) =>
+            metricOnlyInstance
+              .appMetrics({
+                'system.memory.actual.free': 1,
+                'system.cpu.total.norm.pct': 1,
+                'system.memory.total': 1,
+                'system.process.cpu.total.norm.pct': 1,
+              })
+              .timestamp(timestamp)
+          )
+      );
```

Comment:
I don't think it's strictly necessary to do the check for different document types since we are just interested in the existence of _any_ documents

Reply:
I believe a strict check here was necessary given the definition of what exactly for us is data, like different document types. But now we don't need this at all in favour of https://github.com/elastic/kibana/pull/159227#discussion_r1223627308